### PR TITLE
UdpOutput drops message if its size exceeds allowed UDP data size (#1393)

### PR DESCRIPTION
**Pull request by 4r9h**

### CODE REVIEW

**Comment by 4r9h** [comment]
@rafrombrc Hi, could you review?

**Comment by trink** [comment]
This should really be a configuration option and not a constant since the maximum size is network/protocol dependent. With the practical maximum limit of 65507 being a good default for IPv4.

**Comment by 4r9h** [comment]
Good point. I'll update PR to reflect your comment.

**Comment by 4r9h** [comment]
@trink hi, i've updated PR. Could you review it again?

File: plugins/udp/udp_output.go [hunk 1]
```diff
@@ -38,12 +43,16 @@ type UdpOutputConfig struct {
 	Address string
 	// Optional address to use as the local address for the connection.
 	LocalAddress string `toml:"local_address"`
+
+	// Maximum size of message, plugin drops the data if it exceeds this limit.
+	MaxMessageSize int `toml:"max_message_size"`
```
Comment (trink) [inline plugins/udp/udp_output.go]: Should validate the user provided configuration value and have a lower threshold (512 should be reasonable).

File: plugins/udp/udp_output.go [hunk 2]
```diff
@@ -22,6 +22,11 @@ import (
 	"runtime"
 )
 
+const (
```
Comment (trink) [inline plugins/udp/udp_output.go]: Don't really need this since it is only referenced in one place (could just put the value there).

**Comment by 4r9h** [comment]
@trink updated PR again. Added new test for validation of configuration provided by user.

**Comment by trink** [comment]
Thanks, r+

**Comment by rafrombrc** [comment]
This looks awesome, and I'd love to get it into 0.9.1... would you be amenable to closing this PR and opening an identical one against the versions/0.9 branch?

**Comment by 4r9h** [comment]
Sure :)

On Thu, Mar 12, 2015 at 5:44 PM, Rob Miller <notifications@github.com>
wrote:

> This looks awesome, and I'd love to get it into 0.9.1... would you be
> amenable to closing this PR and opening an identical one against the
> versions/0.9 branch?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla-services/heka/pull/1403#issuecomment-78522560>
> .
>



-- 
jarek@reijutsu:~$ fortune
You have Egyptian flu: you're going to be a mummy.
jarek@reijutsu:~$ fortune
You now have Asian Flu.

**Comment by rafrombrc** [comment]
Closed and reopened against versions/0.9 branch in #1416.

**Pull request closed without merging**